Implement "attach-mode"

Leon-Plickat opened the pull request:
Solves #90 

More modes (like "below focus" or "above focus") can be easily added in a future commit.

ifreund commented on river/Output.zig:
```diff
@@ -107,6 +114,8 @@ pub fn init(self: *Self, root: *Root, wlr_output: *c.wlr_output) !void {
 
     self.layout = try std.mem.dupe(util.gpa, u8, "full");
 
+    self.attach_mode = AttachMode.AttachTop;
```
can use an enum literal here
```suggestion
    self.attach_mode = .top;
```

ifreund commented on river/command/attach_mode.zig:
```diff
@@ -0,0 +1,42 @@
+    if (args.len > 2) return Error.TooManyArguments;
+
+    if (std.mem.eql(u8, "top", args[1])) {
+        seat.focused_output.attach_mode = Output.AttachMode.AttachTop;
```
Here as well, can also drop the `Output` import.
```suggestion
        seat.focused_output.attach_mode = .top;
```

ifreund commented on doc/riverctl.1.scd:
```diff
@@ -95,6 +95,9 @@ that tag 1 through 9 are visible.
 ## CONFIGURATION COMMANDS
 
+*attach-mode* *top*|*bottom*
+	Configure where new views should attach into the view stack.
```
```suggestion
	Configure where new views should attach to the view stack for the currently focused output.
```

Leon-Plickat commented:
Maybe it would make more sense to have the mode switching be part of the `viewstack.push()` function instead of different kinds of having multiple push functions and switching every time  a view is attached?

ifreund commented:
> Maybe it would make more sense to have the mode switching be part of the viewstack.push() function instead of having multiple push functions and switching every time a view is attached?

There are other places that `ViewStack.push()` is used (e.g. the zoom command or the focus stack of seats). Putting this logic in a new `ViewStack.attach()` function would make sense though imo.

ifreund commented on river/command/attach_mode.zig:
```diff
@@ -0,0 +1,42 @@
+
+const Error = @import("../command.zig").Error;
+const Seat = @import("../Seat.zig");
+const Output = @import("../Output.zig");
```
```suggestion
```

Leon-Plickat commented:
Renamed `viewstack.push_bottom()` to `viewstack.append()` which is a better name IMO.

Also added `viewstack.attach()`, which now handles the switch. The idea is to call `.attach()` when the way a node is attached can change during runtime and to use `.push()` and the other specific attach methods when the mode is hardcoded anyway for whatever reason (as with the zoom command).
